src/main/java/javax/measure/unit/ProductUnit.java

```diff
@@ -465,7 +532,21 @@ public int getRoot() {
         }
 
         private static final long serialVersionUID = 1L;
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj instanceof Element) {
```

Comment:
Same trick suggestion as above: we can benefit from the internally stored `hashCode` to short-circuit and avoid more expensive computations when we know the hashcode differs and therefore the objects cannot be equal.
```suggestion
            if (obj instanceof Element && obj.hashCode == this.hashCode) {
```